### DISCUSSION TITLE
Raster PMTiles and slider

### DISCUSSION THREAD

**Discussion by vitoborneo** [protomaps/PMTiles#351] [Q&A]
With Leaflet I usually use OSM like basemap and raster tiles like custom map. This is an example:
https://www.borneo.name/cartografia/gsgs/4164/001-passo-del-brennero
I'm trying to do the same with PMTils, but slider and opacity doesn't work.
Vito

**Comment by bdon** [comment]
Please provide information on how to reproduce the problem

**Reply by vitoborneo** [reply]
This is xyz tile page:
https://www.borneo.name/maps/test/index_xyz.html
and this is pmtiles page:
https://www.borneo.name/maps/test/index_pmtiles.html

Regards

**Reply by bdon** [reply]
you are calling `setOpacity` on the PMTiles object, not the leaflet layer.

```
pmtiles.leafletRasterLayer(PMTiles1).addTo(map);
```

```
let layer = pmtiles.leafletRasterLayer(PMTiles1);
layer.addTo(map)
layer.setOpacity(...)
```

**Reply by vitoborneo** [reply]
I updated with your advice, but nothing changed.

**Reply by bdon** [reply]
did you update

```
slide: function (event, ui) {
                layer.setOpacity(ui.value);
            }
```

and 

```
var overlayMaps = {
 "Mappa": layer 
};
```
to properly reference the layer variable?

**Reply by vitoborneo** [reply]
It's no good
https://www.borneo.name/maps/test/index_pmtiles_update.html

**Reply by bdon** [reply]
I'm not able to volunteer helping you debug this JavaScript code, as this doesn't have to do with PMTiles but the usage of Leaflet. Please make sure that your code properly references the layer variable (not the pmtiles object). If you discover that after fixing the JS code there is an issue directly related to PMTiles please add more information.
